toolz/itertoolz.py

```diff
@@ -97,6 +98,32 @@ def groupby(key, seq):
     return rv
 
 
+def indices(*sizes):
+    """ Iterates over a length/shape.
+
+    >>> list(indices(3, 2))
+    [(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1)]
+
+    This can help nicely index an array.
+
+    >>> l = [[1, 2],
+    ...      [3, 4],
+    ...      [5, 6]]
+
+    >>> for i, j in indices(3, 2):
+    ...     print("l[%i][%i] = %i" % (i, j, l[i][j]))
+    l[0][0] = 1
+    l[0][1] = 2
+    l[1][0] = 3
+    l[1][1] = 4
+    l[2][0] = 5
+    l[2][1] = 6
```

Comment:
I guess I am not used to using index access inside for loops, normally people just loop over the values directly and in numpy you don't want to be doing a bunch of scalar accesses like this. To help me understand can you explain some real code that you have written that uses this.